### PR TITLE
Handle namespace not found in replication

### PR DESCRIPTION
**Pull request by yux0**
**What changed?**
1. Handle namespace not found in replication
2. Change client not found error type

**Why?**
1. Drop the replication task if namespace is not found.
2. Return not found error if the client is not found.

**How did you test it?**
Existing tests

**Potential risks**

**Is hotfix candidate?**
No

### CODE REVIEW

File: service/history/replication/task_executor.go
Comment (meiliang86) [inline service/history/replication/task_executor.go:309]: I feel like an existing unit test should fail? Otherwise we probably don't have proper test coverage :)
I stamped this PR but please use your judgement here.

**Review by meiliang86 — APPROVED** [review]

**Comment by meiliang86** [comment]
Also wait for @yycptt to stamp before merge.

File: service/history/replication/task_executor.go
Comment (yux0) [reply service/history/replication/task_executor.go:309]: Let me add one

Comment (yycptt) [reply service/history/replication/task_executor.go:309]: To me it looks like embedding the `shouldProcessTask` logic in the error handling part makes the code a bit hard to understand. 

(FYI, currently the only error returned by GetNamespaceByID is namespace not found, but that may change in the future if we make to make the cache read through. )

Comment (yux0) [reply service/history/replication/task_executor.go]: Revert to original logic and just handle namespace not found error

**Review by yycptt — APPROVED** [review]

**Pull request merged**
